src/main/kotlin/com/cosmotech/connector/adt/impl/AzureDigitalTwinsConnector.kt

```diff
@@ -67,27 +67,30 @@ class AzureDigitalTwinsConnector : Connector<DigitalTwinsClient,List<CsvData>,Li
 
         modelInformationList = AzureDigitalTwinsUtil.retrievePropertiesFromBaseModels(modelInformationList)
 
-        LOGGER.info("Fetching Digital Twin Instances...")
+        LOGGER.info("Fetching Digital Twin Instances Information...")
         val fetchDTInstancesStart = System.nanoTime()
-        val digitalTwinInformation = constructDigitalTwinInstances(modelInformationList, client)
+
+        val digitalTwins = client.query("SELECT * FROM DIGITALTWINS", BasicDigitalTwin::class.java)
+            .sortedBy { it.metadata.modelId }
+        modelInformationList.forEach {
+            val digitalTwinsByModel = digitalTwins.filter { dt -> dt.metadata.modelId == it.id }
```

Comment:
Same issue here. I think filtering the `digitalTwins` collection for each element in `modelInformationList` is expensive too, potentially an `O(n)` operation in the worst case.
A better approach could be to make `digitalTwins` a HashMap, then in the loop access the information right away, e.g.:
```kotlin
val digitalTwins = client.query("SELECT * FROM DIGITALTWINS", BasicDigitalTwin::class.java)
                .groupBy { it.metadata.modelId }
modelInformationList.forEach {
            val digitalTwinsByModel = digitalTwins[it.id]
            //...
}
```
This allows retrieval in constant time.